File: py/checks/components/entrytypes.py
entry_requirements = {
    "Article": {
        "required": ["author", "title", "journaltitle", "date"],
        "recommended": ["doi", "volume", "number", "pages"],
    },
    "Book": {
        "required": ["author", "title", "date", "publisher"],
        "recommended": ["isbn"],
    },
    "Booklet": {
        "required": ["title", "date"],
        "recommended": ["author", "howpublished", "address"],
    },
    "Conference": {  # Alias for Inproceedings
        "required": ["author", "title", "booktitle", "date"],
        "recommended": ["editor", "volume", "number", "series", "pages", "address", "month", "organization", "publisher", "note"],
    },
    "Inproceedings": {  # Given Conference is an alias for Inproceedings, they should have the same requirements
        "required": ["author", "title", "booktitle", "date"],
        "recommended": ["editor", "volume", "number", "series", "pages", "address", "month", "organization", "publisher", "note"],
    },
    "Dataset": {
        "required": ["author", "title", "date", "url", "urldate"],
        "recommended": [],
    },
    "InBook": {
        "required": ["author", "title", "booktitle", "date", "publisher"],
        "recommended": ["volume", "series", "type", "address", "edition", "month", "note"],
    },
    "Manual": {
        "required": ["title"],
        "recommended": ["author", "organization", "address", "edition", "month", "year", "note"],
    },
    "Misc": {
        "required": [],
        "recommended": ["author", "title", "howpublished", "month", "year", "note"],
    },
    "Online": {
        "required": ["author", "title", "date", "url", "urldate"],
        "recommended": [],
    },
    "Report": {
        "required": ["author", "title", "institution", "year"],
        "recommended": ["type", "number", "address", "month", "note"],
    },
    "Thesis": {
        "required": ["author", "title", "school", "year"],
        "recommended": ["type", "address", "month", "note"],
    },
}

# Assigning aliases
entry_requirements["Electronic"] = entry_requirements["Online"]
entry_requirements["Www"] = entry_requirements["Online"]
entry_requirements["Mastersthesis"] = entry_requirements["Thesis"]
entry_requirements["Phdthesis"] = entry_requirements["Thesis"]
entry_requirements["Techreport"] = entry_requirements["Report"]
entry_requirements["Software"] = entry_requirements["Misc"]


def get_requirements():
    return entry_requirements

if __name__ == "__main__":
    # This code will not be executed when imported as a module.
    print("This is the entry_requirements module.")

File: js/components/entrytypes.js
// Define entry types, their required and recommended fields (with all the fields HOGENT prefers! Differs from actual BibLaTeX manual.)
const entryRequirements = {
    Article: {
        required: ['author', 'title', 'journaltitle', 'date'],
        recommended: ['doi', 'volume', 'number', 'pages'],
    },
    Book: {
        required: ['author', 'title', 'date', 'publisher'],
        recommended: ['isbn'],
    },
    Booklet: {
        required: ['author', 'title', 'date', 'publisher'],
        recommended: ['isbn', 'doi', 'url', 'howpublished'],
    },
    Conference: { // Alias for Inproceedings
        required: ['author', 'title', 'booktitle', 'date'],
        recommended: ['editor', 'eventtitle', 'isbn', 'doi', 'url'],
    },
    Dataset: {
        required: ['author', 'title', 'date', 'url', 'urldate'],
        recommended: [],
    },
    InBook: {
        required: ['author', 'title', 'booktitle', 'date', 'publisher'],
        recommended: ['isbn', 'doi', 'pages'],
    },
    Manual: {
        required: ['author', 'title', 'date'],
        recommended: ['organization', 'publisher', 'isbn', 'doi', 'url'],
    },
    Misc: { // Alias for Software
        required: ['author', 'title', 'date'],
        recommended: [],
    },
    Online: { // Alias: Electronic, Www
        required: ['author', 'date', 'title', 'url', 'urldate'],
        recommended: [],
    },
    Report: { // Alias: Techreport
        required: ['author', 'title', 'date', 'type', 'institution'],
        recommended: ['doi', 'url'],
    },
    Thesis: { // Alias: Mastersthesis, Phdthesis
        required: ['author', 'title', 'date', 'type', 'institution'],
        recommended: ['url'],
    },
};

// Assigning aliases
entryRequirements.Electronic = entryRequirements.Online;
entryRequirements.Www = entryRequirements.Online;
entryRequirements.Mastersthesis = entryRequirements.Thesis;
entryRequirements.Phdthesis = entryRequirements.Thesis;
entryRequirements.Techreport = entryRequirements.Report;
entryRequirements.Software = entryRequirements.Misc;
entryRequirements.Inproceedings = entryRequirements.Conference;

// Export the entry requirements object
module.exports = { entryRequirements };
